src/providers/node/mod.rs

```diff
@@ -22,7 +23,8 @@ mod turborepo;
 
 pub const NODE_OVERLAY: &str = "https://github.com/railwayapp/nix-npm-overlay/archive/main.tar.gz";
 
-const DEFAULT_NODE_PKG_NAME: &str = "nodejs-16_x";
+const DEFAULT_NODE_VERSION: u32 = 16;
+const DEFAULT_NODE_PKG_NAME: &str = formatcp!("nodejs-{DEFAULT_NODE_VERSION}_x");
```

Comment:
I don't think we need to have the default node package in String format here, which would reduce the need for the `const_format` dependency. We can instead just convert the `DEFAULT_NODE_VERSION` to a string when we resolve to a pacakge

Reply:
@coffee-cup Cant easily resolve `DEFAULT_NODE_VERSION` to string in tests. So I add a new method to get node version number from pkg name `pkg_to_version_number`, inputs like `nodejs-16_x` outputs will be `16`

Reply:
after ^, we can keep `DEFAULT_NODE_PKG_NAME` and dont need `formatcp`